app/src/main/java/com/example/unlibrary/library/LibraryRepository.java

```diff
@@ -418,4 +425,126 @@ public void declineRequester(String requestedUID, String bookRequestedID, OnSucc
                 })
                 .addOnFailureListener(onDeclineFailure);
     }
+
+    /**
+     * Accepts the selected requester, sets the handoff location, and clears the other requesters
+     *
+     * @param requestedUID      accepted requester user ID
+     * @param bookRequestedID   book ID request is associated with
+     * @param handoffLocation   LatLng of handoff location
+     * @param onSuccessListener code to call on success
+     * @param onFailureListener code to call on failure
+     */
+    public void acceptRequester(String requestedUID, String bookRequestedID, LatLng handoffLocation, OnSuccessListener onSuccessListener, OnFailureListener onFailureListener) {
+        mDb.collection(REQUESTS_COLLECTION)
+                .whereEqualTo(BOOK, bookRequestedID)
+                .whereNotEqualTo(STATE, ARCHIVED)
+                .get()
+                .addOnCompleteListener(requestTask -> {
+                    if (requestTask.isSuccessful()) {
+                        List<Request> requests = requestTask.getResult().toObjects(Request.class);
+                        mDb.runTransaction((Transaction.Function<Void>) transaction -> {
+
+                            // Firestore transactions do not allow writes to occur after a read
+                            DocumentReference acceptedRequestDocument = null;
+                            ArrayList<DocumentReference> declinedRequestDocuments = new ArrayList<>();
+                            for (Request r : requests) {
+                                String id = r.getId();
+                                DocumentReference requestDocument = mDb.collection(REQUESTS_COLLECTION).document(id);
+                                DocumentSnapshot snapshot = transaction.get(requestDocument);
+
+                                if (snapshot.get(REQUESTER).equals(requestedUID)) {
+                                    acceptedRequestDocument = requestDocument;
+
+                                } else {
+                                    declinedRequestDocuments.add(requestDocument);
+                                }
+                            }
+
+                            if (acceptedRequestDocument != null) {
```

Comment:
If acceptedRequestDocument is null, we do not want to continue on the code that is after this if statement. I.e. If we find no request to accept, we should not decline other requests.

I think we should do the opposite check where if acceptedRequestDocument == null, we log and return.

Reply:
I we need something like
```suggestion
                            if (acceptedRequestDocument == null) {
                                onFailureListener.onFailure(new Exception("The request to accept wasn't found"));
                                return;
                            }
```